src/parser/name_resolution.c

```diff
@@ -4836,7 +4868,8 @@ pt_dblink_table_fill_attr_def (PARSER_CONTEXT * parser, PT_NODE * attr_def_node,
   int is_default = 0;
 
   attr_def_node->data_type = NULL;
-  attr_def_node->type_enum = pt_type[attr->type_idx];
+  /* it needs to convert ext type to CCI_U_TYPE */
+  attr_def_node->type_enum = pt_type[(T_CCI_U_TYPE) CCI_GET_COLLECTION_DOMAIN (attr->type_idx)];
```

Comment:
There is a dblink_get_basic_utype() function for a similar purpose. How about using this?
`attr_def_node->type_enum = dblink_get_basic_utype(attr->type_idx)`